src/transformers/models/pix2struct/modeling_pix2struct.py

```diff
@@ -362,7 +362,10 @@ def custom_forward(*inputs):
             hidden_states = layer_outputs[0]
 
             if output_attentions:
-                all_self_attentions = all_self_attentions + (layer_outputs[1],)
+                if len(layer_outputs) >= 3:
+                    all_self_attentions = all_self_attentions + (layer_outputs[2],)
+                else:
+                    all_self_attentions = all_self_attentions + (layer_outputs[1],)
```

Comment:
Do we need this additional if/else logic here? Looking at the code, my understanding is that `output_attentions` is passed to the attention layers. If we've entered this block (`if output_attentions:`) then the layer output should always be: `(attn_output, position_bias, attn_weights)`. 

```suggestion
                all_self_attentions = all_self_attentions + (layer_outputs[2],)
```

Reply:
The issue you mentioned also bothers me. Initially, I thought that if output_attentions is True, the length of layer_outputs would definitely be 3. In fact, I submitted the PR for the first time without if/else, and the code was exactly the same as the one you suggested. However, it didn't pass the tests and reported an index error. This means that sometimes the length of layer_outputs is not 3 but 2. I haven't closely examined the cause of this issue yet, but I think it may be due to a problem with the logic in the original code or with the test cases.

Reply:
OK. Although the indexing is currently wrong, without being confident in the behaviour of the model and these flags, I don't think we should merge in these changes: they make the code harder to understand and are essentially a patch over the continuing bug which should be addressed. 

@younesbelkada, as you added the model, could you look into this?